query_update.go

```diff
@@ -278,6 +278,9 @@ func (q *UpdateQuery) appendSetStruct(
 		if !hasValue && q.omitZero && f.HasZeroValue(model.strct) {
 			continue
 		}
+		if f.SkipUpdate() {
+			continue
+		}
```

Comment:
Perhaps check this as early as possible?

Reply:
Done.